modules/alloy/src/main/ts/ephox/alloy/positioning/layout/LayoutInset.ts

```diff
@@ -19,7 +22,7 @@ const labelPrefix = 'layout-inset';
 const westEdgeX = (anchor: AnchorBox): number => anchor.x;
 
 // returns middle of anchor minus half the element width - used to horizontally centre element to the anchor
-const middleX = (anchor: AnchorBox, element: AnchorElement): number => anchor.x + (anchor.width / 2) - (element.width / 2);
+const middleX = (anchor: AnchorBox, element: AnchorElement, boundsWidth: number): number => anchor.x + (Math.min(anchor.width, boundsWidth) / 2) - (element.width / 2);
```

Comment:
I wonder if there is ever a situation where we want to be sure that the middle is calculated to the middle of the anchor regardless of what the bound is so the old behavior. Since now it's not really a middle since it's clamped by the bounds of the view. So do we need both, like a clampedMiddleX but then you need to special mode all over the place to switch between things is that just over complicating things. But I know to little about the positioning apis to know what this could break.

Reply:
ok, maybe I found a problem, I found a strange behavior in inline editor, that seems better that the previous one, but I need to make some more test

Reply:
ok, it wasn't better in the previous version but the problem seems to resent only when you scroll horizontally and you have an image larger that the visible area, I think it could be something out of scope for this one